File: Lab6/src/main/java/com/example/lab6/MainController.java
package com.example.lab6;

import com.example.lab6.data.DataBaseConnector;
import com.example.lab6.data.DataBaseRepository;
import com.example.lab6.data.Order;
import com.example.lab6.data.Repository;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.fxml.Initializable;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.*;
import javafx.scene.control.cell.PropertyValueFactory;
import javafx.stage.Stage;

import java.io.IOException;
import java.net.URL;
import java.util.List;
import java.util.ResourceBundle;

public class MainController implements Initializable {
    @FXML
    private TableView<Order> listOrders;
    @FXML
    private TableColumn<Order, Integer> id;
    @FXML
    private TableColumn<Order, String> productName;
    @FXML
    private TableColumn<Order, String> deliveryAddress;
    @FXML
    private TableColumn<Order, Integer> amount;

    @FXML
    private TableColumn<Order, Integer> cost;

    @FXML
    private TableColumn<Order, Boolean> prepayment;

    @FXML
    private TableColumn<Order, String> orderDateAndTime;

    @FXML
    TextField deliveryAddressInput;
    @FXML
    ComboBox<String> prepaymentCombo;

    private Repository repository;

    @Override
    public void initialize(URL url, ResourceBundle resourceBundle) {
        repository = new DataBaseRepository(
                new DataBaseConnector("deliveryServiceDB"));

        id.setCellValueFactory(new PropertyValueFactory<>("id"));
        productName.setCellValueFactory(new PropertyValueFactory<>("productName"));
        deliveryAddress.setCellValueFactory(new PropertyValueFactory<>("deliveryAddress"));
        amount.setCellValueFactory(new PropertyValueFactory<>("amount"));
        cost.setCellValueFactory(new PropertyValueFactory<>("cost"));
        prepayment.setCellValueFactory(new PropertyValueFactory<>("prepayment"));
        orderDateAndTime.setCellValueFactory(new PropertyValueFactory<>("orderDateAndTime"));

        prepaymentCombo.getItems().addAll("Усі", "Оплачені", "Неоплачені");
        updateListsView();

    }

    public void updateListsView() {
        List<Order> orders = repository.getAll();
        ObservableList<Order> ordersList = FXCollections.observableList(orders);
        listOrders.setItems(ordersList);
        prepaymentCombo.getSelectionModel().select("Усі");

    }


    @FXML
    public void deleteOrder(ActionEvent actionEvent) {
        Order toDelete = (Order) listOrders.getSelectionModel().getSelectedItem();
        repository.deleteOrder(toDelete.getId());
        updateListsView();
    }

    @FXML
    public void addNewOrder(ActionEvent actionEvent) {
        Stage newWindow = new Stage();
        FXMLLoader loader = new FXMLLoader(DeliveryService.class.getResource(
                "add-order-form.fxml"
        )
        );
        Parent root = null;
        try {
            root = loader.load();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        newWindow.setTitle("Добавити замовлення");
        newWindow.setScene(new Scene(root));
        AddOrderController secondController = loader.getController();
        secondController.set_repository(repository);
        secondController.set_mainController(this);
        newWindow.show();
    }

    @FXML
    public void findOrder(ActionEvent actionEvent) {
        String deliveryAddress = deliveryAddressInput.getText();
        List<Order> orders = null;
        if ("all".equals(deliveryAddress)) {
            orders = repository.getAll();
        } else {
            orders = repository.getAllByDeliveryAddress(deliveryAddress);
        }
        ObservableList<Order> ordersList = FXCollections.observableList(orders);
        listOrders.setItems(ordersList);
    }

    @FXML
    public void filterComboBox(ActionEvent actionEvent) {
        String prepayment = prepaymentCombo.getSelectionModel().getSelectedItem();

        List<Order> orders = null;
        if (prepayment.equals("Усі")) {
            orders = repository.getAll();
        } else if (prepayment.equals("Оплачені")) {
            // Фільтрувати замовлення з переоплатою
            orders = repository.getAllByPrepayment(true);
        } else {
            // Фільтрувати замовлення без передоплати
            orders = repository.getAllByPrepayment(false);
        }
        ObservableList<Order> ordersList = FXCollections.observableList(orders);
        listOrders.setItems(ordersList);
    }

}

File: Lab6Kotlin/src/main/kotlin/com/example/lab6/MainController.kt
package com.example.lab6

import com.example.lab6.data.DataBaseConnector
import com.example.lab6.data.DataBaseRepository
import com.example.lab6.data.Order
import com.example.lab6.data.Repository
import javafx.collections.FXCollections
import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import javafx.fxml.Initializable
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.scene.control.ComboBox
import javafx.scene.control.TableColumn
import javafx.scene.control.TableView
import javafx.scene.control.TextField
import javafx.scene.control.cell.PropertyValueFactory
import javafx.stage.Stage
import java.io.IOException
import java.net.URL
import java.util.*

class MainController : Initializable {
    @FXML
    lateinit var listOrders: TableView<Order>

    @FXML
    lateinit var id: TableColumn<Order, Int>

    @FXML
    lateinit var productName: TableColumn<Order, String>

    @FXML
    lateinit var deliveryAddress: TableColumn<Order, String>

    @FXML
    lateinit var amount: TableColumn<Order, Int>

    @FXML
    lateinit var cost: TableColumn<Order, Int>

    @FXML
    lateinit var prepayment: TableColumn<Order, Boolean>

    @FXML
    lateinit var orderDateAndTime: TableColumn<Order, String>

    @FXML
    lateinit var deliveryAddressInput: TextField

    @FXML
    lateinit var prepaymentCombo: ComboBox<String>
    lateinit var repository: Repository

    override fun initialize(url: URL, resourceBundle: ResourceBundle?) {
        repository = DataBaseRepository(
                DataBaseConnector("deliveryServiceDB"))
        id!!.cellValueFactory = PropertyValueFactory("id")
        productName!!.cellValueFactory = PropertyValueFactory("productName")
        deliveryAddress!!.cellValueFactory = PropertyValueFactory("deliveryAddress")
        amount!!.cellValueFactory = PropertyValueFactory("amount")
        cost!!.cellValueFactory = PropertyValueFactory("cost")
        prepayment!!.cellValueFactory = PropertyValueFactory("prepayment")
        orderDateAndTime!!.cellValueFactory = PropertyValueFactory("orderDateAndTime")
        prepaymentCombo!!.items.addAll("Усі", "Оплачені", "Неоплачені")
        updateListsView()
    }

    fun updateListsView() {
        val orders = repository!!.all
        val ordersList = FXCollections.observableList(orders)
        listOrders!!.items = ordersList
        prepaymentCombo!!.selectionModel.select("Усі")
    }

    @FXML
    fun deleteOrder(actionEvent: ActionEvent?) {
        val toDelete = listOrders!!.selectionModel.selectedItem as Order
        repository!!.deleteOrder(toDelete.id)
        updateListsView()
    }

    @FXML
    fun addNewOrder(actionEvent: ActionEvent?) {
        val newWindow = Stage()
        val loader = FXMLLoader(DeliveryService::class.java.getResource(
                "add-order-form.fxml"
        )
        )
        var root: Parent? = null
        root = try {
            loader.load<Parent>()
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
        newWindow.title = "Добавити замовлення"
        newWindow.scene = Scene(root)
        val secondController: AddOrderController = loader.getController<AddOrderController>()
        secondController.set_repository(repository)
        secondController.set_mainController(this)
        newWindow.show()
    }

    @FXML
    fun findOrder(actionEvent: ActionEvent?) {
        val deliveryAddress = deliveryAddressInput!!.text
        var orders: List<Order>? = null
        orders = if ("all" == deliveryAddress) {
            repository!!.all
        } else {
            repository!!.getAllByDeliveryAddress(deliveryAddress)
        }
        val ordersList = FXCollections.observableList(orders)
        listOrders!!.items = ordersList
    }

    @FXML
    fun filterComboBox(actionEvent: ActionEvent?) {
        val prepayment = prepaymentCombo!!.selectionModel.selectedItem
        var orders: List<Order>? = null
        orders = if (prepayment == "Усі") {
            repository!!.all
        } else if (prepayment == "Оплачені") {
            // Фільтрувати замовлення з переоплатою
            repository!!.getAllByPrepayment(true)
        } else {
            // Фільтрувати замовлення без передоплати
            repository!!.getAllByPrepayment(false)
        }
        val ordersList = FXCollections.observableList(orders)
        listOrders!!.items = ordersList
    }
}